src/PathDrawing/PathDrawingController.cpp

```diff
@@ -0,0 +1,238 @@
+// Copyright eeGeo Ltd (2012-2015), All Rights Reserved
+
+#include "PathDrawingController.h"
+#include "RouteBuilder.h"
+#include "VectorMath.h"
+#include "RouteStyle.h"
+#include "WayPointsFactory.h"
+#include "WayPointsRepository.h"
+#include "WayPointType.h"
+#include "LatLongAltitude.h"
+
+using namespace Eegeo;
+using namespace Eegeo::Routes;
+
+namespace ExampleApp
+{
+    namespace PathDrawing
+    {
+        
+        PathDrawingController::PathDrawingController(Eegeo::Routes::RouteService& routeService
+                                                     , Eegeo::EegeoWorld& eegeoWorld
```

Comment:
Nothing should need a dependency on the entirety of EegeoWorld.  If you only want to update while EegeoWorld is initialised, the current pattern is to tick the update within this conditional in MEA.cpp (https://github.com/eegeo/eegeo-example-app/blob/master/src/MobileExampleApp.cpp#L1115) - This may change in the future to allow apps to subscribe to the initialisation completing, but for now this is consistent with other types